field/src/types.rs

```diff
@@ -490,6 +490,19 @@ pub trait Field64: Field {
     // TODO: Move to `Field`.
     fn from_noncanonical_u64(n: u64) -> Self;
 
+    /// Returns `n` as an element of this field. Assumes that `|n| < Self::ORDER`.
+    // TODO: Should probably be unsafe.
+    // TODO: Move to `Field`.
+    #[inline]
+    fn from_canonical_i64(n: i64) -> Self {
```

Comment:
I’m not sure this method name makes sense. We define canonical representation as anything in {0, …, `ORDER`}, so by that definition a negative number can never be canonical.